packages/flame/lib/src/parallax.dart

```diff
@@ -485,18 +485,18 @@ class Parallax {
     var depth = 0;
     final layers = await Future.wait<ParallaxLayer>(
       dataList.map((data) async {
-        final renderer = await data.load(
-          repeat,
-          alignment,
-          fill,
-          images,
-        );
         final velocityMultiplier =
             List.filled(depth, velocityDelta).fold<Vector2>(
           velocityDelta,
           (previousValue, delta) => previousValue.clone()..multiply(delta),
         );
         ++depth;
+        final renderer = await data.load(
```

Comment:
how come this fixes the issue? You just moved the code, am I missing something?

Reply:
Because now it gets its velocity before the loading of the image, so the time it takes to load the image won't have any effect on which velocity it gets.

Reply:
Since the function is async it will continue with the other layers when it has started the loading, so layers out of order could get their velocity before an earlier layer, if that makes it clearer?

Reply:
but nothing inside this closure mutates anything on the outer scope, the only thing outside is `velocityDelta`, which is final (or no modified). It shouldn't matter if you await on the beginning or on the end of the method.

Reply:
It does, it mutates `depth`, which is used while calculating the `velocityMultiplier`.

Reply:
Made it cleaner, so that it uses `mapIndexed` instead.